### PR TITLE
[BarPlot] - Auto width calculation takes border in account

### PR DESCRIPTION
**Pull request by bluong**
Auto width currently makes sure that none of the bars overlap.  Now it makes sure that it doesn't extend over the edge

[@bluong 1/29 14:34] EDIT: If user does not specify the barWidth, then this will be used to gate the barWidths at the edge of the scale.

### CODE REVIEW

File: src/components/plots/barPlot.ts [hunk 1]
```diff
@@ -388,7 +388,20 @@ export module Plot {
...
+        if (this._barAlignmentFactor !== 0 && minScaledDatum > 0) {
+          barPixelWidth = Math.min(barPixelWidth, minScaledDatum / this._barAlignmentFactor);
+        }
+        var maxScaledDatum = _Util.Methods.max(scaledData, 0);
+        if (this._barAlignmentFactor !== 1 && maxScaledDatum < barWidthDimension) {
+          var margin = barWidthDimension - maxScaledDatum;
+          barPixelWidth = Math.min(barPixelWidth, margin / (1 - this._barAlignmentFactor));
+        }
+
+        barPixelWidth *= 0.95;
```
Comment (jtlan) [inline src/components/plots/barPlot.ts]: Is it possible to store off the `0.4` and `0.95` somewhere?

**Comment by jtlan** [comment]
[minor] magic number storage

File: src/components/plots/barPlot.ts (hunk 1 above)
Comment (bluong) [inline src/components/plots/barPlot.ts]: Yep.  WIll change

**Comment by jtlan** [comment]
Err, I meant pull them out as member/static variables, something where they're accessible outside of the method....

**Comment by bluong** [comment]
I consdereed pulling them out as member static variables but my first thought is they shouldn't be considered global... Do we want them to be?

**Comment by crmorford** [comment]
Note: Same data, very different appearance
Before:
screen shot 2015-02-05 at 2 06 51 pm
After:
screen shot 2015-02-05 at 2 07 00 pm

That being said, looks good. I just like recording this stuff somewhere.

**Pull request merged**